airflow/contrib/operators/gcp_transfer_operator.py

```diff
@@ -0,0 +1,856 @@
+    :param object_conditions: Optional transfer service object conditions; see
+        https://cloud.google.com/storage-transfer/docs/reference/rest/v1/TransferSpec
+    :type object_conditions: dict
+    :param transfer_options: Optional transfer service transfer options; see
+        https://cloud.google.com/storage-transfer/docs/reference/rest/v1/TransferSpec
+    :type transfer_options: dict
+    :param wait: Wait for transfer to finish
+    :type wait: bool
+    """
+
+    template_fields = ('gcp_conn_id', 's3_bucket', 'gcs_bucket', 'description',
+                       'object_conditions')
+    ui_color = '#e09411'
+
+    @apply_defaults
+    def __init__(self,
+                 s3_bucket,
+                 gcs_bucket,
+                 project_id=None,
+                 aws_conn_id='aws_default',
+                 gcp_conn_id='google_cloud_default',
+                 delegate_to=None,
+                 description=None,
+                 schedule=None,
+                 object_conditions=None,
+                 transfer_options=None,
+                 wait=True,
+                 *args,
+                 **kwargs):
+
+        super(S3ToGoogleCloudStorageTransferOperator, self).__init__(
+            *args,
+            **kwargs)
+        self.s3_bucket = s3_bucket
+        self.gcs_bucket = gcs_bucket
+        self.project_id = project_id
+        self.aws_conn_id = aws_conn_id
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+        self.description = description
+        self.schedule = schedule
+        self.object_conditions = object_conditions
+        self.transfer_options = transfer_options
+        self.wait = wait
+
+    def execute(self, context):
+        transfer_hook = GCPTransferServiceHook(
+            gcp_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to)
+        body = {
+            "description": self.description,
+            "status": GcpTransferJobsStatus.ENABLED,
+            "transferSpec": {
+                'awsS3DataSource': {
+                    'bucketName': self.s3_bucket,
+                },
+                'gcsDataSink': {
+                    'bucketName': self.gcs_bucket,
+                },
+            }
+        }
+
+        if self.project_id is not None:
+            body["project_id"] = self.project_id
+
+        if self.schedule is not None:
+            body['schedule'] = self.schedule
+
+        if self.object_conditions is not None:
+            body['transferSpec']['object_conditions'] = self.object_conditions
+
+        if self.transfer_options is not None:
+            body['transferSpec']['transferOptions'] = self.transfer_options
+
+        TransferJobPreprocessor(
+            body=body,
+            aws_conn_id=self.aws_conn_id
+        ).process_body()
+
+        job = transfer_hook.create_transfer_job(
+            body=body
+        )
+
+        if self.wait:
+            transfer_hook.wait_for_transfer_job(job)
```

Comment:
I think we should return more information here in Xcom. Let's discuss it.

Reply:
This is not an area of our current work. This operator was previously created by another programmer. Maybe we should add it to future work? 
 
I agree that changes in this area would be useful. I agree that changes in this area would be useful, but they should have a global scope, not a single one.

Reply:
As discussed - that's fine. let's leave it as it is for now